### ISSUE TITLE
Opening the share modal

### ISSUE THREAD

**Issue by cyberbaloo** [suitenumerique/docs#2002]
## Bug Report

**Problematic behavior**
The rendering of the share modal opening is not vocalized when navigating with a screen reader.

**Expected behavior/code**
When the share button is activated, the modal opens, the focus must move to the modal and announce its opening

**Steps to Reproduce**
1. Log in to a document in read mode
2. Enable the screen reader
3. Navigate to the share button with the keyboard
4. And then the bug happens!

**Critères RGAA**
Critère 7.1 : [**Chaque script est-il, si nécessaire, compatible avec les technologies d’assistance ?**](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#7.1)

**Impact** 
The lack of vocalization when the modal opens is a non-conformity because the focus must be moved to the modal or its first focusable element upon opening, and assistive technologies must convey the change of context.

This is the most critical case because the focus remains positioned on the modal opening button and the user has no way of knowing that the modal has just opened despite their action.

**Additional context/Screenshots**
<img width="1104" height="874" alt="Image" src="https://github.com/user-attachments/assets/d16f9785-65af-4d6b-8e29-a562dd37503b" />

**Comment by Ovgodd** [comment]
Hello @cyberbaloo for the modal announcement it will be fixed here : https://github.com/suitenumerique/docs/pull/2007
the SR announce  now :

`Partager  bouton  `
then : 

`Partager le document  dialogue`
and auto focus on the search input : 
`Résultats de la recherche  liste déroulante  développé`
